### DISCUSSION TITLE
V to C translation?

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#20707] [Ideas]
We got C to V but we don't have V to C translation. It would be very cool and more popular if the Vlang could also do V to C translation. It'll help programmers to understand more deeper in embedded programming. 🤤

**Comment by JalonSolov** [comment]
`v -o file.c file.v` will give you the "V to C" translation.  In other words, V will simply output the generated C code, without calling the C compiler to create the executable.  The key is the `-o file.c`.

Warning - this file will include everything needed to turn your code into a full executable, so will probably be a lot more than you wanted.

You can also use the `-printfn <module__fn>` option to only see the C code for a particular function, such as `-printfn main__main` to see your `fn main` code.

**Reply by ghost** [reply]
Wow, that's very very impressive. Wish they tell about it more in their videos. It's very fun and exciting to know that V can do both translations even if it's not perfect.  Cheers, bro. Also, I don't know when V will reach version 1. Can I except V will reach version 1 before the start of 2026?

**Reply by JalonSolov** [reply]
We don't do dates.  v1.0 will be released when it's ready.  I do expect it before 2026, but we'll have to wait and see, as V still has more work to be done before 1.0 is ready.

**Reply by ghost** [reply]
Thanks.
